pytorch_lightning/trainer/trainer.py

```diff
@@ -2096,15 +2098,12 @@ def model(self, model: torch.nn.Module) -> None:
 
     @property
     def log_dir(self) -> Optional[str]:
-        if self.logger is None:
-            dirpath = self.default_root_dir
-        elif isinstance(self.logger, TensorBoardLogger):
-            dirpath = self.logger.log_dir
-        elif isinstance(self.logger, LoggerCollection):
+        if len(self.loggers) != 1 or isinstance(self.logger, LoggerCollection):
```

Comment:
When creating a LoggerCollection of size 1, the log_dir property returns default_root_dir, however we can't easily substitute this with trainer.loggers because trainer.loggers can't distinguish between a list of length 1 and a single logger. Both are stored as a list of length 1. Currently I'm checking trainer.logger, but this won't work once trainer.logger and LoggerCollection are deprecated. Should this be handled differently?